common/src/stack/report-system/command/report/system/fixtures/disk_partition_check.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import paramiko
+import socket
+import re
+from collections import namedtuple
+from itertools import groupby
+
+# Takes in a hostname, returns the lsblk data in a dictionary of disks with a nested
+# dictionary of partitions per disk which each has a named tuple of the name, mountpoint,
+# disk label, file system type, and disk size. Returns an empty dictionary if the host
+# cannot be ssh into or lsblk does not return any data
+def get_partition_data(hostname):
+
+	disks = {}
+	partitions = {}
+
+	backend_ssh = paramiko.SSHClient()
+	backend_ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+	# Skip the test if we can't ssh in
+	try:
+			backend_ssh.connect(hostname=f'{hostname}')
+
+	except (paramiko.SSHException, socket.error):
+		return {}
+
+	# Get backend host partition informaiton using lsblk
+	stdin, stdout, stderr = backend_ssh.exec_command('lsblk -r -n -o name,mountpoint,label,fstype,size')
+
+	lsblk_data = stdout.readlines()
+
+	# Turn each key value string pair into it's own entry into the list
+	if lsblk_data:
+		host_part_data = ' '.join(lsblk_data).replace('\n', '').split(' ')
+
+	else:
+		return {}
+
+	# Convert to a list of named tuples for each partition consisting
+	# of the partition name, label, and mountpoint
+	for part in zip(
+
+		# Iterate through each partition which is seperated every 5 elements
+		# in the list of lsblk output
+		host_part_data[::5], host_part_data[1::5], host_part_data[2::5],
+		host_part_data[3::5], host_part_data[4::5]
+	):
+		part_name = part[0]
+		part_tuple = namedtuple(part_name, ['name', 'mountpoint', 'label', 'fstype', 'size'])
```

Comment:
You're use of `nametuple` isn't quite right. Above your for loop, you create a name tuple like:
```
Partition = namedtuple("Partition", ['name', 'mountpoint', 'label', 'fstype', 'size'])
```
Then in the loop you do:
```
part_tuple = Partition(part_name, part[1], part[2], part[3], part[4])
```
Thank of the `namedtuple` call as constructing a Class on the fly.